File: day1/main.js
import * as readline from "node:readline";
import * as fs from "fs";

const fileStream = fs.createReadStream("input.txt");

const rl = readline.createInterface({
  input: fileStream,
});
let calories = [];
let highest = 0;
let totalCalories = [];
rl.on("line", (line) => {
  if (line.match(/^$/)) {
    let sum = 0;
    for (let i = 0; i < calories.length; i++) {
      sum += calories[i];
    }
    totalCalories.push(sum);
    if (sum > highest) {
      highest = sum;
    }
    calories = [];
  }
  if (line != "") calories.push(+line);
});

rl.on("close", () => {
  let secondHighest;
  let thirdHighest;

  for (let i = 0; i < totalCalories.length; i++) {
    if (
      !secondHighest ||
      (totalCalories[i] > secondHighest && totalCalories[i] != highest)
    ) {
      secondHighest = totalCalories[i];
    }
  }
  for (let i = 0; i < totalCalories.length; i++) {
    if (
      !thirdHighest ||
      (totalCalories[i] > thirdHighest &&
        totalCalories[i] != highest &&
        totalCalories[i] != secondHighest)
    ) {
      thirdHighest = totalCalories[i];
    }
  }
  console.log({ highest, secondHighest, thirdHighest });
  console.log({ total: highest + secondHighest + thirdHighest });
});

File: day1/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	data, err := os.Open("input.txt")
	if err != nil {
		panic(err)
	}
	scan := bufio.NewScanner(data)
	lineSlice := []int{}
	sum := []int{}
	var highest int
	for scan.Scan() {
		line, _ := strconv.ParseInt(scan.Text(), 10, 64)
		if line == 0 {
			var total int
			for _, item := range sum {
				total += item
			}
			if total > highest {
				highest = total
			}
			lineSlice = append(lineSlice, total)
			sum = nil
			continue
		}
		sum = append(sum, int(line))
	}
	var secondHighest, thirdHighest int
	for _, item := range lineSlice {
		if item != highest && item > secondHighest {
			secondHighest = item
		}
	}
	for _, item := range lineSlice {
		if item != highest && item != secondHighest && item > thirdHighest {
			thirdHighest = item
		}
	}

	// fmt.Println(highest, secondHighest, thirdHighest)
	fmt.Println(highest + secondHighest + thirdHighest)
}
